src/Calculator.ts

```diff
@@ -345,3 +339,196 @@ function addSDE(report: R4.IMeasureReport, measureURL: string, result: Execution
+    return 0.5 * (sorted[sorted.length / 2 - 1] + sorted[sorted.length / 2]);
+  } else {
+    return sorted[sorted.length / 2];
+  }
+}
+
+function populationTotal(group: R4.IMeasureReport_Group, type: PopulationType) {
+  return (
+    group.population?.find(pop => {
+      return pop.code?.coding && pop.code.coding.length > 0 && pop.code?.coding[0].code === type;
+    })?.count || 0.0
+  );
+}
+
+function createMeasureScore(
+  measure: R4.IMeasure,
+  group: R4.IMeasureReport_Group,
+  detail: DetailedPopulationGroupResult,
+  report: R4.IMeasureReport
+) {
+  const scoringCode =
+    measure.scoring?.coding?.find(c => c.system === 'http://hl7.org/fhir/measure-scoring')?.code || '';
+  switch (scoringCode) {
+    case MeasureScoreType.PROP:
+      // (Numerator - Numerator Exclusions) / (Denominator - D Exclusions - D Exceptions).
+      const numeratorCount =
+        populationTotal(group, PopulationType.NUMER) - populationTotal(group, PopulationType.NUMEX);
+      const denominatorCount =
+        populationTotal(group, PopulationType.DENOM) -
+        populationTotal(group, PopulationType.DENEX) -
+        populationTotal(group, PopulationType.DENEXCEP);
+
+      group.measureScore = {
+        // TODO: what if value for denominator 0? ... do we need to subtract denex, dexecep... probably, as https://ecqi.healthit.gov/system/files/eCQM-Logic-and-Guidance-2018-0504.pdf
+        value: denominatorCount === 0 ? 0 : (numeratorCount / denominatorCount) * 1.0
+      };
+      break;
+    case MeasureScoreType.CV:
+      let observations = [];
+      if (detail.episodeResults) {
+        // find all episode results with a true measure observation
+        const relevantEpisodes =
+          detail.episodeResults?.filter(
+            er => er.populationResults.find(pr => pr.populationType === PopulationType.OBSERV)?.result
+          ) || [];
+
+        // Note: only uses first of potentially multiple observations in each episode (since we can't tell which observation is which)
+        // if there are multiple oservations, then this may cause inconsistent behavior
+        observations = relevantEpisodes.map(
+          er => er.populationResults.find(pr => pr.populationType === PopulationType.OBSERV)?.observations[0]
+        );
+      } else {
+        // CV for patient-based results is untested
+        const obsResultPop = detail.populationResults?.find(pr => pr.populationType === PopulationType.OBSERV);
+        observations = obsResultPop?.result ? [obsResultPop.observations[0]] : [];
+      }
+
+      // find aggregation type
+      const measureGroup = measure.group?.find(g => g.id === group.id);
+      const observPop = measureGroup?.population?.find(p =>
+        p.code?.coding?.find(c => c.code === 'measure-observation')
+      );
+      const aggregation =
+        observPop?.extension?.find(
+          e => e.url === 'http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-aggregateMethod'
+        )?.valueCode || '';
+
```

Comment:
Same here, could throw an error if no aggregate method is defined?

Reply:
I added the error under the default for the aggregation method switch (which is what will happen with an empty string), so this will catch both unfound and unexpected aggregation methods